### PR TITLE
[Keymap] Add new IIDX gamepad keymap for Synth Labs Solo

### PR DESCRIPTION
**Pull request by hongaaronc**
## Description

These changes add another keymap, which makes this macropad function as a gamepad which works out-of-the-box with the game [beatmania IIDX INFINITAS](https://p.eagate.573.jp/game/infinitas/2/index.html).

I have tested this keymap manually in-game, and verified that the behavior works as expected. Please also refer to the new README to see what the expected behaviors are.

The rotary encoder in this keymap is intended to have slightly different behavior from the most standard rotary encoder implementation:
 - In the most rotary encoder implementations, spinning an encoder continuously generally outputs repeated taps of a keycode
 - In this implementation, spinning an encoder continuously outputs a keycode being pressed and held until the spinning stops
 - Additionally, if a spinning encoder's direction changes, release the keycode corresponding to the previous direction immediately

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition ~~or update~~)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/synthlabs/solo/keymaps/iidx/keymap.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+}
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    encoder_action_register(index, clockwise);
+    return false;
+}
+
+#endif
```
Comment (drashna) [inline keyboards/synthlabs/solo/keymaps/iidx/keymap.c:91]: Use the encoder map feature instead. 
https://docs.qmk.fm/#/feature_encoders?id=encoder-map

```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/synthlabs/solo/keymaps/iidx/keymap.c (hunk 1 above)
Comment (hongaaronc) [reply keyboards/synthlabs/solo/keymaps/iidx/keymap.c]: Updated it to use the encoder map feature - that does simplify it a bit. Thanks!

Also updated the PR description to be more clear about the intended encoder behavior for this keymap.

File: keyboards/synthlabs/solo/keymaps/iidx/keymap.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            unregister_joystick_button(CCW_JOYSTICK_BUTTON);
+            register_joystick_button(CW_JOYSTICK_BUTTON);
+            encoder_last_update_time = timer_read();  /* Update the last time that the encoder was detected as rotated */
+            return false;
+        default:
+            return true;
+    }
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/synthlabs/solo/keymaps/iidx/keymap.c:54]:
```suggestion
void housekeeping_task_user(void) {
```

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by hongaaronc** [comment]
Thanks for looking over the changes and for the approvals. Looks like I don’t have authorization to merge though, so could someone help merge them for me?

**Comment by hongaaronc** [comment]
Thanks!

**Pull request merged**